packages/suite/src/components/suite/graph/TransactionsGraph/TransactionsGraph.tsx

```
@@ -90,6 +91,76 @@ export interface FiatGraphProps extends CommonProps {
 
 export type TransactionsGraphProps = CryptoGraphProps | FiatGraphProps;
 
+const emptyList: ReturnType<typeof selectAccountTransactionsWithNulls>[] = [];
+const useGraphUpdater = ({
+    shouldUpdateGraph,
+    account,
+}: {
+    shouldUpdateGraph: (abortController: AbortController) => Promise<any> | undefined;
+    account: Account | undefined;
+}) => {
+    const [currentPromise, setCurrentPromise] = useState<{
+        promiseId: string;
+        txIds: string[];
+        promise: Promise<any>;
+        abortController: AbortController;
+    } | null>(null);
+
+    const allTransactions = useSelector(state =>
+        account ? selectAccountTransactionsWithNulls(state, account.key) : emptyList,
+    );
+
+    const newestTransactions = allTransactions
+        .slice(0, 3)
+        .flatMap(tx => tx)
+        .filter((tx): tx is WalletAccountTransaction =>
+            Boolean(Boolean(tx) && tx && !isPending(tx)),
+        );
+
+    const txIds = newestTransactions.map(tx => tx.txid);
+
+    const promiseId = txIds.join('-');
+
+    useEffect(() => {
+        if (promiseId !== currentPromise?.promiseId && account) {
+            const nextAbortController = new AbortController();
+
+            currentPromise?.abortController.abort();
+
+            setCurrentPromise({
+                promiseId,
+                abortController: nextAbortController,
+                txIds,
+                promise: Promise.resolve()
+                    .then(() =>
+                        currentPromise?.promise?.then(
+                            result => result,
+                            error => {
+                                console.warn(
+                                    'useGraphUpdater: Previous graph update promise failed',
+                                    error,
+                                );
+                            },
+                        ),
+                    )
```

this is just for debugging?

---

like the logging? yes, I probably delete it

---

but like I still want to "swallow" this error here and keep going to the next promise

---

ok then